pkg/node/volumemgr.go

```diff
@@ -289,11 +289,16 @@ func (m *VolumeManager) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 			}
 		}
 	} else {
+		// need to handle volume CR delete request
 		switch volume.Spec.CSIStatus {
 		case apiV1.Created:
 			volume.Spec.CSIStatus = apiV1.Removing
 			ll.Debug("Change volume status from Created to Removing")
 		case apiV1.Removing:
+		// Failed drive shouldn't be cleaned up - might cause DU
+		case apiV1.Failed:
```

Comment:
Does it make sense to perform the same actions as for REMOVED volumes here too?

Reply:
Refactored